yt_dlp/extractor/mixch.py

```diff
@@ -48,8 +57,20 @@ def _real_extract(self, url):
                 'protocol': 'm3u8',
             }],
             'is_live': True,
+            '__post_extractor': self.extract_comments(video_id=video_id),
         }
 
+    def _get_comments(self, video_id):
+        yield from traverse_obj(self._download_json(
+            f'https://mixch.tv/api-web/lives/{video_id}/messages', video_id,
+            note='Downloading comments', errnote='Failed to download comments'), (..., {
+                'author': ('name', {str_or_none}),
+                'author_id': ('user_id', {int_or_none}),
+                'id': ('message_id', {str_or_none}),
+                'text': ('body', {str_or_none}),
+                'timestamp': ('created', {int_or_none}),
```

Comment:
* It is better to have `id=None` than `id=""`.
* `author_id` should be returned as as a `str`, not `int`.
* `{TYPE_or_none}` converts the data into `TYPE` while `{TYPE}` verifies the data is `TYPE` without converting

Considering these, I believe this is the proper impl:
```suggestion
                'author': ('name', {str}),
                'author_id': ('user_id', {str_or_none}),
                'id': ('message_id', {str}, {lambda x: x or None),
                'text': ('body', {str}),
                'timestamp': ('created', {int}),
```
Correct?

Reply:
Thanks! I must have been confused "{TYPE_or_none}" and "{TYPE}" at that time.

Fixed in 5bbd6c18b8a01e2f8966e8f4d586b6ffd4f689ae.